pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/RangeIndexCreator.java

```diff
@@ -0,0 +1,471 @@
+          break;
+        default:
+          throw new UnsupportedOperationException("Range index is not supported for columns of data type:" + valueType);
+      }
+
+      //Stores the docId - this will be sorted based on the values in FORWARD INDEX in the end
+      _docIdValueBuffer = createTempBuffer((long) _numValues * Integer.BYTES, _forwardIndexValueBufferFile);
+      _docIdBuffer = new IntValueBuffer(_docIdValueBuffer);
+    } catch (Exception e) {
+      destroyBuffer(_forwardIndexValueBuffer, _forwardIndexValueBufferFile);
+      destroyBuffer(_forwardIndexValueBuffer, _docIdBufferFileForSorting);
+      throw e;
+    }
+  }
+
+  @Override
+  public void add(int dictId) {
+    _valueBuffer.put(_nextDocId, dictId);
+    _docIdBuffer.put(_nextDocId, _nextDocId);
+    _nextDocId = _nextDocId + 1;
+  }
+
+  @Override
+  public void add(int[] dictIds, int length) {
+    for (int i = 0; i < length; i++) {
+      int dictId = dictIds[i];
+      _valueBuffer.put(_nextDocId, dictId);
+      _docIdBuffer.put(_nextValueId, _nextDocId);
+      _nextValueId = _nextValueId + 1;
+    }
+  }
+
+  @Override
+  public void addDoc(Object document, int docIdCounter) {
+    throw new IllegalStateException("Bitmap inverted index creator does not support Object type currently");
+  }
+
+  @Override
+  public void seal()
+      throws IOException {
+    //sort the forward index copy
+    //go over the sorted value to compute ranges
+    IntComparator comparator = (i, j) -> {
+      Number val1 = _valueBuffer.get(_docIdBuffer.get(i).intValue());
+      Number val2 = _valueBuffer.get(_docIdBuffer.get(j).intValue());
+      return _valueBuffer.compare(val1, val2);
+    };
+    Swapper swapper = (i, j) -> {
+      Number temp = _docIdBuffer.get(i).intValue();
+      _docIdBuffer.put(i, _docIdBuffer.get(j).intValue());
+      _docIdBuffer.put(j, temp);
+    };
+    Arrays.quickSort(0, _numValues, comparator, swapper);
+
+//    dump();
+    //FIND THE RANGES
+    List<Pair<Integer, Integer>> ranges = new ArrayList<>();
+
+    int boundary = _numDocsPerRange;
+    int start = 0;
+    for (int i = 0; i < _numValues; i++) {
+      if (i > start + boundary) {
+        if (comparator.compare(i, i - 1) != 0) {
+          ranges.add(new Pair(start, i - 1));
+          start = i;
+        }
+      }
+    }
+    ranges.add(new Pair(start, _numValues - 1));
+
+    // Create bitmaps from inverted index buffers and serialize them to file
+    //HEADER
+    //   # VERSION (INT)
+    //   # DATA_TYPE (String -> INT (length) (ACTUAL BYTES)
+    //   # OF RANGES (INT)
+    //   <RANGE START VALUE BUFFER> #R & ValueSize
+    //   Range Start 0
+    //    .........
+    //   Range Start R - 1
+    //   Bitmap for Range 0 Start Offset
+    //      .....
+    //   Bitmap for Range R Start Offset
+    //BODY
+    //   Bitmap for range 0
+    //   Bitmap for range 2
+    //    ......
+    //   Bitmap for range R - 1
+    long bytesWritten = 0;
```

Comment:
I don't see _forwardIndexValueBuffer being used

Reply:
why dont need that right?